src/Polly.Core/Simmy/Outcomes/OutcomeChaosStrategy.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Polly.Telemetry;
+
+namespace Polly.Simmy.Outcomes;
+
+#pragma warning disable S3928 // Custom ArgumentNullException message
+
+internal class OutcomeChaosStrategy<T> : MonkeyStrategy<T>
+{
+    private readonly ResilienceStrategyTelemetry _telemetry;
+
+    public OutcomeChaosStrategy(OutcomeStrategyOptions<Exception> options, ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        if (options.Outcome is null && options.OutcomeGenerator is null)
+        {
+            throw new ArgumentNullException(nameof(options.Outcome), "Either Outcome or OutcomeGenerator is required.");
+        }
+
+        _telemetry = telemetry;
+        Fault = options.Outcome;
+        OnFaultInjected = options.OnOutcomeInjected;
+        FaultGenerator = options.OutcomeGenerator is not null ? options.OutcomeGenerator : (_) => new(options.Outcome);
+    }
+
+    public OutcomeChaosStrategy(OutcomeStrategyOptions<T> options, ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        if (options.Outcome is null && options.OutcomeGenerator is null)
+        {
+            throw new ArgumentNullException(nameof(options.Outcome), "Either Outcome or OutcomeGenerator is required.");
+        }
+
+        _telemetry = telemetry;
+        Outcome = options.Outcome;
+        OnOutcomeInjected = options.OnOutcomeInjected;
+        OutcomeGenerator = options.OutcomeGenerator is not null ? options.OutcomeGenerator : (_) => new(options.Outcome);
+    }
+
+    public Func<OnOutcomeInjectedArguments<T>, ValueTask>? OnOutcomeInjected { get; }
+
+    public Func<OnOutcomeInjectedArguments<Exception>, ValueTask>? OnFaultInjected { get; }
+
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<T>?>>? OutcomeGenerator { get; }
+
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<Exception>?>>? FaultGenerator { get; }
+
+    public Outcome<T>? Outcome { get; private set; }
```

Comment:
```suggestion
    public Outcome<T>? Outcome { get; }
```

to indicate it's read-only